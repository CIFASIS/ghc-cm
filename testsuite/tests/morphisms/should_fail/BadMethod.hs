module BadMethod where

class A a where f :: a -> String

class B a where g :: a -> Int

class morphism A -> B where
  g a = f a 

