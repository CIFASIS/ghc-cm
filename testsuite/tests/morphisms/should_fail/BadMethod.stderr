
BadMethod.hs:8:9: error:
    • Couldn't match type ‘[Char]’ with ‘Int’
      Expected type: Int
        Actual type: String
    • In the expression: f a
      In an equation for ‘g’: g a = f a
      In the morphism declaration
        ‘class morphism A -> B -- Defined at BadMethod.hs:7:1’
